herddb-core/src/main/java/herddb/core/StreamDataScanner.java

```diff
@@ -72,7 +74,11 @@ public DataAccessor next() throws DataScannerException {
 
     @Override
     public void close() throws DataScannerException {
+        if (closed.compareAndSet(false, true)) {
+            if (transaction != null) {
+                transaction.decreaseRefCount();
+            }
+        }
         super.close();
```

Comment:
Can we invoke even super.close() under closed.compareAndSet?

Reply:
done